palindrome-products/palindrome-products_test.spec.js

```diff
@@ -7,7 +7,7 @@ describe("Palindrome", function() {
     var largest = palindromes.largest;
 
     expect(largest.value).toEqual(9);
-    expect([1, 9]).toEqual(largest.factors);
+    expect([[3, 3], [1, 9]], [[1, 9], [3, 3]]).toContain(largest.factors);
   });
 
   xit("largets palindrome from double digit factors", function() {
```

Comment:
No, no, no. The test passes. But it's still different from the JS one. That's why I think the ECMAScript solution is not 100% correct.

The ECMAScript solution (link to the [example.js](https://github.com/exercism/xecmascript/blob/master/palindrome-products/example.js) file) must be different to the JS one (link to the [example.js](https://github.com/exercism/xjavascript/blob/master/palindrome-products/example.js)) because `largest.factors` are different. JS tests look very similar to Ruby tests, so I assume the JS tests and solution are correct. Then, the ECMAScript solution might not be 100% correct. It's not your fault, the ECMAScript solution code was there before you catch the mistake in the tests.

In order to get the solution right, I was proposing to copy & paste the JS code and then refactor it to make to more ECMAScript 2015 style.